### PR TITLE
allow usage behind an HTTP Proxy

### PR DESCRIPTION
**Pull request by simonlopez**
This fixes the case where the server is not able to do direct http request and needs to use a proxy.

### CODE REVIEW

**Comment by lonvia** [comment]
There was a different solution proposed quite some time ago in issue #33. Did you have a look at that?

**Comment by simonlopez** [comment]
@lonvia I didn't foud the issue before, sorry. I've made changes to use @dhalbert proposal but with clean settings vars and support for proxy for https connexion if needed in future.

File: settings/settings.php [hunk 1]
```diff
@@ -102,3 +109,26 @@ function user_busy_cost()
...
 	@define('CONST_Log_File_SearchLog', '');
 	@define('CONST_Log_File_ReverseLog', '');
+
+	// handle http proxy when using file_get_contents
+	if (CONST_HTTP_Proxy) {
+		$proxy = 'tcp://' . CONST_HTTP_Proxy_Host . ':' . CONST_HTTP_Proxy_Port;
+		$aHeaders = array();
+		if(CONST_HTTP_Proxy_Login != null && CONST_HTTP_Proxy_Login != '' && CONST_HTTP_Proxy_Password != null && CONST_HTTP_Proxy_Password != '') {
+			$auth = base64_encode(CONST_HTTP_Proxy_Login . ':' . CONST_HTTP_Proxy_Password);
+			$aHeaders = array("Proxy-Authorization: Basic $auth", "Authorization: Basic $auth");
```
Comment (lonvia) [inline settings/settings.php]: Is the Authorization header correct here? I'd expect that Proxy-Authorization is sufficient.

**Comment by lonvia** [comment]
Thanks, that is so much nicer than a commented-out section. Two tiny cosmetic changes would be nice before that can be merged: remove the empty line change in `lib/lib.php` and move the section that configures the proxy into `lib/init-cmd.php`.

**Comment by simonlopez** [comment]
@lonvia , I've made the suggested changes. Should be ok now.

**Pull request merged**